packages/cli/internal/pkg/cli/util/timeout.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"errors"
+	"time"
+)
+
+const (
+	deploymentTimeout = 30 * time.Minute
+)
+
+func DeployWithTimeout(timeoutFunction func() error) error {
+	// channel to mark when a deployment successfully completes
+	completionChannel := make(chan error)
+	go func() {
+		err := timeoutFunction()
+		completionChannel <- err
+	}()
+	select {
+	case err := <-completionChannel:
+		return err
+	case <-time.After(deploymentTimeout):
+		return errors.New("deployment taking longer then expected. please review stack deployment in cloudformation")
```

Comment:
nit: "*the* deployment *is* taking longer than expected. *P*lease review *the* stack in *C*loud*F*ormation"